### DISCUSSION TITLE
Simple HTML comment breaks re-render

### DISCUSSION THREAD

**Discussion by Musamba24** [livewire/livewire#5142] [7. Bugs (archived)] [answered]
Honestly i don't know if it's just a strange combinations of things or it's general.
I don't even know if it's actually a bug, if not i'm sorry for the useless post.

After hours of debugging i found out that a single comment line (Probably added by PhpStorm as i started to use it since few weeks) **BEFORE** the component's root element breaks the page after **ANY** request, here's the code: (I removed anything unnecessary, it was a pretty big file)

```
<!--suppress ALL -->
<div id="wrapper">
	<a onclick="$('#importArticlesFileModal').modal('show');" type="button" class="btn btn-lg btn-warning" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Importa articoli">
		<i class="fa-solid fa-cloud-arrow-up" aria-hidden="true"></i>
	</a>
	<div wire:ignore.self class="modal fade" id="importArticlesFileModal" tabindex="-1" aria-labelledby="importArticlesFileModalLabel" aria-hidden="true">
		<div class="modal-dialog modal-dialog-centered">
			<div class="modal-content">
				<div class="modal-body">
					<input id='excelFile' class="form-control form-control-lg" type="file" wire:model="excelFile">
				</div>
				<div class="modal-footer">
					<button type="button" class="btn btn-lg btn-secondary" data-bs-dismiss="modal">Chiudi</button>
				</div>
			</div>
		</div>
	</div>

    <div id="content">
        {{-- TABLE --}}
        <div class="mt-2">
            <livewire:storage.articles.articles-table />
        </div>
    </div>
</div>
```

Just removing the 1st line fixes the page.
After opening the page and the modal it's going anything ok but as soon i select a file (Or do any other operation inside it) all elements inside the component disappear leaving the modal's backdrop effect.

![Schermata del 2022-07-31 19-18-54](https://user-images.githubusercontent.com/83073028/182038046-823fb992-9ecf-442c-95bf-d237c428c9fa.png)

**Comment by joshhanley** [comment, marked as answer]
@Musamba24 yeah that's correct, Livewire requires one root element to function correctly.

But you currently have two (a comment and a div). See here in the docs https://laravel-livewire.com/docs/2.x/troubleshooting#root-element-issues

Hence why Livewire is breaking 🙂

Hope this helps!

**Comment by muath-ye** [comment]
I faced this issue , it only need to have one root html element without any HTML comment before

```blade
<div>
    {{-- Success is as dangerous as failure. --}}
</div>
```
